src/libraries/System.Security.Cryptography.Algorithms/tests/DsaEcDsaSignatureFormatTests.cs

```diff
@@ -0,0 +1,1166 @@
+            }
+            else
+            {
+                Assert.ThrowsAny<CryptographicException>(action);
+            }
+        }
+
+        protected void AssertThrowsForNullDataOrHash(Action action)
+        {
+            AssertExtensions.Throws<ArgumentNullException>(DataOrHashArgumentName(), action);
+        }
+
+        [Theory]
+        [MemberData(nameof(XunitTestCases))]
+        public void SignatureEncodedAsIeeeP1363CanBeVerified(TestDataInfo testInfo)
+        {
+            TestData test = GetTestData(testInfo);
+            byte[] signature = Sign(test, DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
+            Assert.True(signature.Length % 2 == 0, "Signature length must be an even number");
+            Assert.True(Verify(test, signature, DSASignatureFormat.IeeeP1363FixedFieldConcatenation));
+        }
+
+        [Theory]
+        [MemberData(nameof(XunitTestCases))]
+        public void SignatureEncodedAsRfc3279CanBeVerified(TestDataInfo testInfo)
+        {
+            TestData test = GetTestData(testInfo);
+            byte[] signature = Sign(test, DSASignatureFormat.Rfc3279DerSequence);
+
+            // First byte must ASN.1 sequence tag (0x30)
+            Assert.Equal(0x30, signature[0]);
+
+            Assert.True(Verify(test, signature, DSASignatureFormat.Rfc3279DerSequence));
+        }
+
+        [Theory]
+        [InlineData(DSASignatureFormat.IeeeP1363FixedFieldConcatenation)]
+        [InlineData(DSASignatureFormat.Rfc3279DerSequence)]
+        public void ValidOffsetAndCountSignsAndVerifies(DSASignatureFormat signatureFormat)
+        {
+            if (!IsAcceptingOffsetAndLength())
+            {
+                return;
+            }
+
+            Assert.True(IsOperatingOnData());
+
+            TestData test = AnyTestCase(minLength: 5);
+            byte[] data = test.Data;
+
+            void SetData(bool useByteArray, int offset, int count)
+            {
+                if (useByteArray)
+                {
+                    test.Data = data.AsSpan(offset, count).ToArray();
+                    test.DataOffset = 0;
+                    test.DataLength = count;
+                }
+                else
+                {
+                    test.Data = data;
+                    test.DataOffset = offset;
+                    test.DataLength = count;
+                }
+            }
+
+            (int, int)[] offsetsAndCounts = new (int, int)[]
+            {
+                (1, data.Length - 1),
+                (1, 3)
+            };
+
+            foreach ((int offset, int count) in offsetsAndCounts)
+            {
+                SetData(!IsSigner(), offset, count);
+
+                if (!IsSigner())
+                {
+                    test.UpdateSignature(signatureFormat);
+                }
+
+                byte[] sig = Sign(test, signatureFormat);
+
+                SetData(IsSigner(), offset, count);
+                Assert.True(Verify(test, sig, signatureFormat));
```

Comment:
This test failed once... but we don't know which walk through a foreach or why.  Consider upgrading all of the Assert.True (or Assert.False) calls to use the message parameter to report what they're testing.  Remember that the message is displayed as

```
Message
    Expected: True
    Actual: False
```

So this one would be

```C#
Assert.True(
    Verify(test, sig, signatureFormat),
    $"Signature verifies for Slice({offset}, {count}) when IsSigner is {IsSigner()}");
```